ingestion_service/producer/core/routers/api_endpoints_input.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# DISCLAIMER: This software is provided "as is" without any warranty,
+# express or implied, including but not limited to the warranties of
+# merchantability, fitness for a particular purpose, and non-infringement.
+#
+# In no event shall the authors or copyright holders be liable for any
+# claim, damages, or other liability, whether in an action of contract,
+# tort, or otherwise, arising from, out of, or in connection with the
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
+# @Author  : Tek Raj Chhetri
+# @Email   : tekraj@mit.edu
+# @Web     : https://tekrajchhetri.com/
+# @File    : api_endpoints_input.py
+# @Software: PyCharm
+
+from fastapi import APIRouter, HTTPException, Body, Depends
+from fastapi import File, Form, UploadFile
+from fastapi.responses import JSONResponse
+from core.configure_rabbit_mq import publish_message
+import logging
+from core.file_validator import validate_file_extension, validate_mime_type
+import json
+from core.pydantic_schema import InputJSONSLdchema, InputJSONSchema, InputTextSchema
+from core.shared import is_valid_jsonld
+from typing import Annotated
+from core.models.user import LoginUserIn
+from core.security import get_current_user, require_scopes
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+
+@router.post("/ingest/file", summary="Ingest a either CSV, JSON, EXCEL, PDF, RDF, TTL, JSONLD or TEXT files",
+             dependencies=[Depends(require_scopes(["write"]))])
+async def ingest_file(user: Annotated[LoginUserIn, Depends(get_current_user)],
+                      id: str = Form(...),
+                      type: str = "file",
+                      posting_user: str = Form(...),
+                      filename: str = Form(...),
+                      file: UploadFile = File(...)):
+    logger.info("Started ingestion operation")
+
+    if not validate_mime_type(file.content_type):
+        raise HTTPException(status_code=400,
+                            detail="Only CSV, JSON, EXCEL, PDF, RDF, TTL, JSONLD and TEXT files are supported.")
+
+    if not validate_file_extension(file.filename):
+        raise HTTPException(status_code=400,
+                            detail="Unsupported file extension. Supported extensions: csv, json, xls, txt and pdf")
+
+    content = await file.read()
+    publish_message(content)
+    logger.info("Successful ingestion operation")
+    return JSONResponse(
+        content={"message": "File uploaded successfully", "id": id, "user": posting_user, "type":{type}, "filename": filename})
+
+
+@router.post("/ingest/raw/json", dependencies=[Depends(require_scopes(["write"]))])
+async def ingest_json(user: Annotated[LoginUserIn, Depends(get_current_user)],
```

Comment:
There is no await call here -- is this truly async?

Reply:
same comment as above.